create_builder.go

```diff
@@ -65,7 +68,7 @@ func (c *Client) CreateBuilder(ctx context.Context, opts CreateBuilderOptions) e
 		return errors.Wrap(err, "setting lifecycle")
 	}
 
-	for _, b := range opts.Config.Buildpacks.Buildpacks() {
+	for i, b := range opts.Config.Buildpacks.Buildpacks() {
```

Comment:
Separation between `Buildpacks` and `Packages` seem to be becoming less valuable. We probably want to consider creating a more meaningful abstractions here. The various considerations should be about how buildpacks are retrieved and how buildpacks are added. In other words, the term "Package" is just one format type of many supported.

Reply:
This point is great, and I totally agree, particularly given that I was differentiating a new form of Package (a file, vs an image).